tensorboard/plugins/core/core_plugin.py

```diff
@@ -88,14 +91,47 @@ def get_plugin_apps(self):
         '/histograms': self._redirect_to_index,
         '/images': self._redirect_to_index,
     }
-    if self._assets_zip_provider:
-      with self._assets_zip_provider() as fp:
-        with zipfile.ZipFile(fp) as zip_:
-          for path in zip_.namelist():
-            gzipped_asset_bytes = _gzip(zip_.read(path))
-            apps['/' + path] = functools.partial(
+    apps.update(self.get_resource_apps())
+    return apps
+
+  def get_resource_apps(self):
+    apps = {}
+    if not self._assets_zip_provider:
+      return apps
+
+    with self._assets_zip_provider() as fp:
+      with zipfile.ZipFile(fp) as zip_:
+        # First collect the shasums that will later be used. Since hashes are
```

Comment:
Optional, but might be a little more straightforward to just loop over the HTML files we want to serve, and for each one, derive the shasum file name and read it then?  Then it's more like the shasums are a specific extra step we do for each HTML file, rather than a bunch of work we do up front no matter what.